lib/Serialization/Deserialization.cpp

```diff
@@ -4132,7 +4124,20 @@ llvm::Error DeclDeserializer::deserializeDeclAttributes() {
     unsigned recordID = MF.fatalIfUnexpected(
         MF.DeclTypeCursor.readRecord(entry.ID, scratch, &blobData));
 
-    if (isDeclAttrRecord(recordID)) {
+    if (recordID == ERROR_FLAG) {
+      IsInvalid = true;
```

Comment:
Is it useful to have an assertion that `ERROR_FLAG` is not seen multiple times for the same decl?

Reply:
Is there a case you're thinking of that the assertion would catch? Doesn't seem overly useful to me, but I suppose it would imply *something* is wrong with the module?

Reply:
I was considering it as a sanity check to ensure that `writeDeclErrorFlag()` is not called multiple times by accident during serialization.